string namespace
string query
---
string[] values
string error
